emcc.py

```diff
@@ -761,16 +761,7 @@ def phase_setup(options, state, newargs):
         else:
           message = arg + ': Unknown format, not a static library!'
         exit_with_error(message)
-      if file_suffix in DYNAMICLIB_ENDINGS and not building.is_bitcode(arg) and not building.is_wasm(arg):
-        # For shared libraries that are neither bitcode nor wasm, assuming its local native
-        # library and attempt to find a library by the same name in our own library path.
-        # TODO(sbc): Do we really need this feature?  See test_other.py:test_local_link
-        libname = removeprefix(get_library_basename(arg), 'lib')
-        flag = '-l' + libname
-        diagnostics.warning('map-unrecognized-libraries', f'unrecognized file type: `{arg}`.  Mapping to `{flag}` and hoping for the best')
-        state.add_link_flag(i, flag)
-      else:
-        input_files.append((i, arg))
```

Comment:
Perhaps we could add an error here that suggests the proper form, to help people update? That is, if `/abs/path/libfoo.so` does not exist, but `libfoo.a` exists in the sysroot, we could say `Absolute path .. provided, but the contents there are not valid for wasm. Did you mean -lfoo ?`

Reply:
I'd rather not for the following reasons:

1. I think its unlikely anyone is actually relying on this/
2. I think the `unknown file format` is pretty reasonable and actionable
3. Do a search of all the library paths at this point is tricky.. especially given the we sometime build libraries on demand and they might not exist yet.   
4. Its hard to recommend `-lfoo` just because somebody passed `somepath/libfoo.so`.   In most cases the solution to "unknown file format somepath/libfoo.so" is to rebuild `somepath/libfoo.so` with `emcc` not to replace that with `-lfoo`

Reply:
Fair enough, lgtm.